sample_size/metrics.py

```diff
@@ -31,6 +34,10 @@ def check_positive(number: float, name: str) -> float:
         else:
             return number
 
+    @abstractmethod
+    def generate_p_value(self, true_alt: bool, size: int, variants: int) -> Any:
```

Comment:
I can't remember: Did we say we were going to make `variants` an instance attribute of metrics, or repeat each metric in the list of metrics?

Reply:
We were certain about not repeating each metric for `variants-1` times in metrics registration, but we don't have strong preference on whether we should make `variants` an instance attribute or just an input parameter of `generate_p_value`. I went with the latter one, because to me `variants` is similar to experiment-level parameters like alpha and power, and there's no need to make them as each metric class's instance attribute